godtools/App/Share/Data/FavoritedResourcesRepository/Cache/RealmFavoritedResourcesCache.swift

```diff
@@ -64,6 +64,14 @@ class RealmFavoritedResourcesCache {
             .map({FavoritedResourceDataModel(realmFavoritedResource: $0)})
     }
     
+    func getFavoritedResourcesSortedByPosition(ascendingOrder: Bool = true) -> [FavoritedResourceDataModel] {
```

Comment:
Hey @rachaelblue I notice this method doesn't run through the validation, only the publisher does.  And with this method exposed as internal it could be called and the list wouldn't be sorted.  I would suggest moving the validation to this method and the publisher call this method.